.github/workflows/deploy.yml

```
@@ -12,6 +12,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: '1.77'
```

[dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain)のREADMEによると、`toolchain`を明示的に指定する場合はActionのブランチを`@master`とする必要がありそうです？

> Rustup toolchain specifier e.g. stable, nightly, 1.42.0, nightly-2022-01-01. Important: the default is to match the `@rev` as described above. When passing an explicit toolchain as an input instead of `@rev`, you'll want to use "dtolnay/rust-toolchain@master" as the revision of the action.

```suggestion
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: '1.77'
```